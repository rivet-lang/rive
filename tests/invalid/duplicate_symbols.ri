fn main() {
    let x = 1;
    let x = 1; // FAIL

    let y = 2;
    {
        let y = 5; // disallow variable shadowing
    }
}

fn other(a: i32) {
    let a = 1;
}

fn other(a: i32) { // FAIL
    let a = 1;
}

class RuntimeError : Error { }
class RuntimeError : Error { } // FAIL

struct Gota { }
struct Gota { } // FAIL

enum State {
    Idle,
    Idle // FAIL
}
