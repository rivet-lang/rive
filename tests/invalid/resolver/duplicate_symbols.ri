fn main() {
    let x = 1;
    let x = 1;

    let y = 2;
    {
        let y = 5; // disallow variable shadowing
    }


    let L1 = 2;
L1:

L2:
    let L2 = true;

    let m = {
    L3:
        let L3 = 1;
    };
}

fn other(a: i32) {
    let a = 1;
}

errtype RuntimeError;
errtype RuntimeError;

struct Gota { }
struct Gota { }

mod module {
    struct App { }
    struct App { }
}

// extend `module` mod
mod module {
    struct Game {
        art: bool;
        art: bool;
    }
}

union Obj {
    i32,
    i32
}

enum State {
    Idle,
    Idle
}

extend module::Game { }
