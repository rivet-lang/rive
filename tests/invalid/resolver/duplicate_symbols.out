tests/failing/resolver/duplicate_symbols.ri:28:9: error: another symbol with this name already exists: `RuntimeError`
     |
  28 | errtype RuntimeError;
     |         ^
tests/failing/resolver/duplicate_symbols.ri:31:8: error: another symbol with this name already exists: `Gota`
     |
  31 | struct Gota { }
     |        ^
tests/failing/resolver/duplicate_symbols.ri:35:12: error: another symbol with this name already exists: `App`
     |
  35 |     struct App { }
     |            ^
tests/failing/resolver/duplicate_symbols.ri:42:9: error: field `art` is already declared
     |
  42 |         art: bool;
     |         ^
tests/failing/resolver/duplicate_symbols.ri:46:7: error: union `Obj` has duplicate variant type `i32`
     |
  46 | union Obj {
     |       ^
tests/failing/resolver/duplicate_symbols.ri:51:6: error: enum `State` has duplicate variant `Idle`
     |
  51 | enum State {
     |      ^
tests/failing/resolver/duplicate_symbols.ri:56:8: error: cannot extend non-local types
     |
  56 | extend module::Game { }
     |        ^
rivetc: error: could not compile package `core`, aborting due to 7 previous errors